website/static/js/components/socialshare.js

```diff
@@ -0,0 +1,52 @@
+/**
+* Mithril components for social sharing.
+*/
+'use strict';
+var $ = require('jquery');
+var m = require('mithril');
+
+var utils = require('js/components/utils');
+var required = utils.required;
+
+var ShareButtons = {
+    openLinkInPopup: function(href) {
+        window.open(href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes,width=600,height=400');
+        return false;
+    },
+    view: function(ctrl, options) {
+        var file_name = encodeURIComponent(required(options, 'file_name'));
+        var share_url = encodeURIComponent(required(options, 'share_url'));
```

Comment:
Minor: `share_url` is a bit redundant (this is a ShareButtons component); how about just `url`?